libraries/core_libs/network/src/network.cpp

```diff
@@ -162,14 +159,14 @@ std::shared_ptr<network::tarcap::TaraxaPeer> Network::getPeer(dev::p2p::NodeID c
   return taraxa_capability_->getPeersState()->getPeer(id);
 }
 
-void Network::sendPbftBlock(dev::p2p::NodeID const &id, PbftBlock const &pbft_block, uint64_t const &pbft_chain_size) {
+void Network::sendPbftBlock(const dev::p2p::NodeID &id, const PbftBlock &pbft_block, uint64_t pbft_chain_size) {
   LOG(log_dg_) << "Network send PBFT block: " << pbft_block.getBlockHash() << " to: " << id;
   taraxa_capability_->sendPbftBlock(id, pbft_block, pbft_chain_size);
 }
 
-void Network::sendPbftVote(dev::p2p::NodeID const &id, std::shared_ptr<Vote> const &vote) {
-  LOG(log_tr_) << "Network sent PBFT vote: " << vote->getHash() << " to: " << id;
-  taraxa_capability_->sendPbftVote(id, vote);
+const std::shared_ptr<taraxa::network::tarcap::PacketHandler> &Network::getSpecificHandler(
```

Comment:
Try to use template for getSpecificHandler method in all classes - we might get rid of those pointer casts that we have to use everywhere:

Methodin network and tarcap would look like this:

```
template <typename PacketHandlerType>
const std::shared_ptr<PacketHandlerType>& getSpecificHandler() {
  return taraxa_capability_->getSpecificHandler<PacketHandlerType>();
} 
```

Method in PacketHandlers;

```
template <typename PacketHandlerType>
const std::shared_ptr<PacketHandlerType>& getSpecificHandler() {
  return static_pointer_cast<PacketHandlerType> .....
} 
```

Reply:
If change like this:

/Users/qigao/Taraxa/taraxa-node/libraries/core_libs/network/src/network.cpp:169:30: error: no matching member function for call to 'getSpecificHandler'
  return taraxa_capability_->getSpecificHandler(packet_type);
         ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~
/Users/qigao/Taraxa/taraxa-node/libraries/core_libs/network/include/network/tarcap/taraxa_capability.hpp:61:45: note: candidate template ignored: couldn't infer template argument 'PacketHandlerType'
  const std::shared_ptr<PacketHandlerType> &getSpecificHandler(SubprotocolPacketType packet_type);

"PacketHandlerType" cannot been deduced, it's in non-deduced context

Reply:
I create another PRhttps://github.com/Taraxa-project/taraxa-node/pull/1736 for this. Could you please take a look?

Reply:
yeah because you are calling it wrong - this is not how template functions work. Nevermind, dont use this idea